docs/index.md

```diff
@@ -0,0 +1,184 @@
+- `query` (string): If the available options are not sufficient, you can also define a custom query to select the data that should be dumped
+
+## Filters / Conditions
+
+There are a number of filters that you can use to reduce the amount of dumped data. Most filters are based on standard SQL conditions.
+
+A filter is an array with three items: [operator, column, value].
+
+Your filters can then look like this:
+```yaml
+    filters:
+        - [in, 'id', [1, 2, 3]] # only select rows which id is 1, 2 or 3
+        - [like, 'name', 'Markus %'] # only select rows which column "name" starts with Markus
+```
+
+Currently, the following operators are available:
+- eq
+- neq
+- lt
+- lte
+- gt
+- gte
+- isNull
+- isNotNull
+- like
+- notLike
+- in
+- notIn
+
+## Converters
+
+There are lots of data converts available that help to convert any sensitive information into arbitrary generated data. You can define converters for a table under the `converters` key. Each column can have multiple converters which are applied in sequence.
+
+The configuration looks similar to the following:
+
+```yaml
+    converters:
+        name:
+            - JohnDoe
+        street:
+            - StreetName
+        company_name:
+            - Faker:
+                formatter: company
+
+```
+
+You can find all available Converters [here](https://github.com/digilist/SnakeDumper/tree/master/src/Converter). In the future I might provide a list with examples for each converter, but currently there is none. So please take a look at the converters to see if they accept parameters.
+
+By using the faker converter you can use any available data generator that is part of the famous [Faker library](https://github.com/fzaninotto/Faker).
+
+## Relations between tables & referential integrity
+
+It is typical for a relational database to have relations between tables. For example, you can have customers and billings. If you know exclude some customers you do not want to have the billings of those customers included in your dump.
+
+Consider the following example:
+
+Table `customers`:
+
+ id | first_name | last_name
+---: | ---: | ---:
+1 | Markus | Fasselt
+2 | Linus | Torvalds
+3 | Rasmus | Lerdorf
+
+Table `billings`:
+
+id | customer_id | amount
+---: | ---: | ---:
+1 | 1 | 20
+2 | 1 | 30
+3 | 2 | 10
+4 | 3 | 10
+5 | 3 | 42
+6 | 3 | 20
+
+When you now define, that you only want to have the first to customers (id 1 and 2), the result should be the following:
+
+Table `customers`:
+
+id | first_name | last_name
+---: | ---: | ---:
+1 | Markus | Fasselt
+2 | Linus | Torvalds
+
+Table `billings`:
+
+id | customer_id | amount
+---: | ---: | ---:
+1 | 1 | 20
+2 | 1 | 30
+3 | 2 | 10
+
+As you can see, customer 3 is not included in the customers table. Furthermore, the billings do not contain the billings of customer 3.
```

Comment:
So the SnakeDumper will resolve relations out of the box?
There is no need to configure these relations?

Reply:
Yes, that's correct. Sorry if you did that :see_no_evil: